src/compiler/html/inline-esm-import.ts

```diff
@@ -56,50 +74,85 @@ export const optimizeEsmImport = async (
   return false;
 };
 
+/**
+ * Update relative module specifiers to instead be nested inside of a supplied
+ * directory, transforming e.g. all imports of the form `'./foo.js'` to
+ * `'/build/foo.js'`.
+ *
+ * @param code the code to transform
+ * @param newDir the directory which should be prepended to all module
+ * specifiers in the code
+ * @returns a manifest containing transformed code and a list of transformed
+ * module specifiers
+ */
 export const updateImportPaths = (code: string, newDir: string) => {
-  const orgImportPaths: string[] = [];
-  const tsSourceFile = ts.createSourceFile('module.ts', code, ts.ScriptTarget.Latest);
-  ts.transform(tsSourceFile, [readImportPaths(orgImportPaths)]);
+  const orgModulePaths = readModulePaths(code);
```

Comment:
I made a change here because I don't think it's actually necessary to use `ts.transform` here, we were doing so purely to gather information about the source code, not to actually run a transformer and then emit code from it. So instead we can just create a `ts.SourceFile` and then iterate through the `statements` on it

Reply:
Agreed - I can't think of/see a reason to use `transform` here 🤔

Reply:
I thought about doing the lean-more-into-TypeScript refactor to actually emit the code but I think that's a _lot_ more overhead actually